elasticsearch/src/main/java/org/hibernate/search/elasticsearch/util/impl/GsonHttpEntity.java

```diff
@@ -135,7 +135,8 @@ public GsonHttpEntity(Gson gson, List<JsonObject> bodyParts) {
 		Objects.requireNonNull( bodyParts );
 		this.gson = gson;
 		this.bodyParts = bodyParts;
-		this.contentLength = attemptOnePassEncoding();
+		this.contentLength = -1;
+		attemptOnePassEncoding();
```

Comment:
I can live  this change since you seem to love it, but IMO it's looking horrible :)
it's a function and you're refactoring it into updating `contentLenght` by side effect _maybe_?

I do realize that several other state fields in this class are mutated as side effect by working on the various buffers and progress pointers, but I'd still try to avoid such things whenever possible.

Reply:
> it's a function and you're refactoring it into updating contentLenght by side effect _maybe_?

Yeah, that's what "attempt" means ;)

> I do realize that several other state fields in this class are mutated as side effect by working on the various buffers and progress pointers, but I'd still try to avoid such things whenever possible.

Sure, it there was an easy way to avoid side-effect completely, I would. But since there is not, I'd rather keep the side-effects in the same method, so that the purpose of the method is clearer.

But I'm not sure we'll be able to agree on that one :)

Reply:
ok. I didn't mean it in negative way, just pointing out that such changes are a bit subjective and sometimes like to better understand the thought process.
I guess what puts me off is the explicit assign to `-1`, which we might change our mind about in the next line. But I'll take it :)